src/community/nsg-profile/src/main/resources/applicationContext.xml

```diff
@@ -1,19 +1,57 @@
-                           http://www.springframework.org/schema/beans/spring-beans.xsd">
-  <bean id="versioning" class="org.geoserver.nsg.versioning.TimeVersioningCallback">
-    <constructor-arg ref="catalog"/>
-  </bean>
-  <!--bean id="versioningLayerConfig" class="org.geoserver.web.publish.LayerConfigurationPanelInfo">
-    <property name="id" value="wfsVersioningConfig"/>
-    <property name="titleKey" value="data.resource.config.wfs"/>
-    <property name="componentClass" value="WfsVersioningConfig"/>
-    <property name="supportedTypes">
-      <list>
-        <value>org.geoserver.catalog.FeatureTypeInfo</value>
-      </list>
-    </property>
-  </bean-->
-</beans>
+<beans xmlns="http://www.springframework.org/schema/beans" xmlns:task="http://www.springframework.org/schema/task" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd   http://www.springframework.org/schema/task http://www.springframework.org/schema/task/spring-task.xsd">
+	<bean class="org.geoserver.nsg.versioning.TimeVersioningCallback" id="versioning">
+		<constructor-arg ref="catalog"/>
+	</bean>
+	<!--bean class="org.geoserver.web.publish.LayerConfigurationPanelInfo" id="versioningLayerConfig">
+		<property name="id" value="wfsVersioningConfig"/>
+		<property name="titleKey" value="data.resource.config.wfs"/>
+		<property name="componentClass" value="WfsVersioningConfig"/>
+		<property name="supportedTypes">
+			<list>
+				<value>org.geoserver.catalog.FeatureTypeInfo</value>
+			</list>
+		</property>
+	</bean-->
+	<bean class="org.geoserver.nsg.pagination.random.IndexConfiguration" id="indexConfiguration"/>
+	<!-- Index result dispatcher callback -->
+	<bean class="org.geoserver.nsg.pagination.random.IndexResultTypeDispatcherCallback" id="indexResultTypeDispatcherCallback">
+		<description>When a request using the index result type comes in a
+			dispatcher callback
+			this switch the &quot;index&quot; value by the &quot;hits&quot; value
+		</description>
+		<constructor-arg ref="geoServer"/>
+		<constructor-arg ref="indexConfiguration"/>
+	</bean>
+	<!-- Page result dispatcher callback -->
+	<bean class="org.geoserver.nsg.pagination.random.PageResultsDispatcherCallback" id="pageResultsDispatcherCallback">
+		<description>The PageResults operation will allow clients to query
+			random positions of an existing result set (stored GetFeature
+			request) that was previously created using the index result type
+		</description>
+		<constructor-arg ref="geoServer"/>
+	</bean>
+	<!-- Page result service -->
+	<bean class="org.geoserver.nsg.pagination.random.PageResultsWebFeatureService" id="pageResultsTarget">
+		<constructor-arg ref="geoServer"/>
+		<constructor-arg ref="indexConfiguration"/>
+	</bean>
+	<bean class="org.geoserver.platform.Service" id="wfsServicePageResults">
```

Comment:
Confused, how does this work? It acts as a separate service that triggers only when version 2.0.2 is used? But doesn't this prevent to have 2.0.2 implemented in core? (which eventually we might want to do, even if today it's not a problem, it declares version 2.0).